netbox_onboarding/worker.py

```diff
@@ -29,43 +30,58 @@ def onboard_device(task_id, credentials):
     """Process a single OnboardingTask instance."""
     username = credentials.username
     password = credentials.password
+    secret = credentials.secret
 
-    try:
-        ot = OnboardingTask.objects.get(id=task_id)
-    except OnboardingTask.DoesNotExist:
-        # TODO: maybe we started before the DB was done writing it, or maybe it was deleted out from under us?
-        time.sleep(1)
-        ot = OnboardingTask.objects.get(id=task_id)
+    ot = OnboardingTask.objects.get(id=task_id)
```

Comment:
Did some aspect of the refactoring make it so we can no longer encounter the `OnboardingTask.DoesNotExist` race condition?

Reply:
I have never experienced this race-condition in thousands of onboardingtasks, do we have any data collected to track this and propose deterministic solution instead of guessed sleep time ?

Reply:
I know it's something I encountered more than once, early in developing this plugin. I haven't seen it in some time, which I took to mean that this hack was doing its job. I'm hesitant to remove it unless we're fixing some more pressing issue by doing so.

Reply:
Per our discussion, I'm okay with this change - hopefully if there's any issue here it'll be uncovered during pre-release testing for 2.0.